app/src/main/java/com/training/app/trainingapp/main/view/forget_password/ForgetPasswordViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.training.app.trainingapp.main.view.forget_password
+
+import androidx.core.util.PatternsCompat
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import javax.inject.Inject
+
+class ForgetPasswordViewModel  @Inject constructor() : ViewModel() {
+
+    private val _userEmail = MutableStateFlow("")
+    val userEmail: StateFlow<String> = _userEmail
+
+    private val _emailValidateState = MutableStateFlow(false)
+    val emailValidateState: StateFlow<Boolean> = _emailValidateState
+    fun validateEmail() {
+        _emailValidateState.value = PatternsCompat.EMAIL_ADDRESS.matcher(_userEmail.value).matches()
+    }
+
+    fun onEmailChanged(email: String) {
+        _userEmail.value = email
+    }
+
+    fun onSubmitButtonClicked() {
+        validateEmail()
+        if (_emailValidateState.value) {
+            sendEmailForPasswordRecovery()
+        }
+    }
+
+    private fun sendEmailForPasswordRecovery() {
```

Comment:
I think we don need this in current state and we should add it in another issue

Reply:
I think it can stay because no implementation has been done for it, it is only specified here what is going to be done when the button is clicked.

Reply:
At least add a //todo comment if there is no implementation.